pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -97,30 +120,63 @@ func (rc *remoteClient) setConfig(watchCtx context.Context, kubeconfig []byte) e
 	}
 	rc.client = remoteClient
 
-	newWatcher, err := rc.client.Watch(watchCtx, &kueue.WorkloadList{})
+	err = rc.startWatcher(watchCtx, kueue.GroupVersion.WithKind("Workload").GroupKind().String(), &workloadKueueWatcher{})
 	if err != nil {
-		return nil
+		return err
+	}
+
+	// add a watch for all the adapters implementing multiKueueWatcher
+	for kind, adapter := range adapters {
+		watcher, implementsWatcher := adapter.(multiKueueWatcher)
+		if !implementsWatcher {
+			continue
+		}
+		err := rc.startWatcher(watchCtx, kind, watcher)
+		if err != nil {
+			// not being able to setup a watcher is not ideal but we can function
+			// with only thw wl watcher, just log the error.
+			ctrl.LoggerFrom(watchCtx).V(2).Error(err, "unaable to start the watcher", "kind", kind)
+		}
+	}
+
+	rc.kubeconfig = kubeconfig
+	return nil
+}
+
+func (rc *remoteClient) startWatcher(ctx context.Context, kind string, w multiKueueWatcher) error {
+	log := ctrl.LoggerFrom(ctx).WithValues("watchKind", kind)
```

Comment:
When doing the watcher for jobs please use a label to filter out jobs not-owner by MultiKueue. The analogous approach as we do for GC. This could be the same label reused. IMO no need for a different label key for Job and workload, because the intention is the same. Same label keys can be generally used on different objects.

Reply:
I see, but this means we may actually get events from another MultiKueue instance sharing worker, or other jobs using the prebuilt-workload. I would suggest to add a follow up to align this and use the origin label later on.

Reply:
I'll alingn them in the last pr  I marge (this or GC)

Reply:
sgtm